### PR TITLE
revert status_message to original code

### PR DESCRIPTION
**Pull request by vanHoesel**
After long thinking and going back and forth ... we should fix this immediately, before the world collapses

HTTP::Status status_message MUST return `undef` for it was the ONLY way to tell if a status code is registered at IANA.

any other suggestions are always welcome

### CODE REVIEW

File: lib/HTTP/Status.pm [hunk 1]
```diff
@@ -126,38 +126,29 @@ our %EXPORT_TAGS = (
...
-        || $_[0] == 301    # Moved Permanently
-        || $_[0] == 404    # Not Found
-        || $_[0] == 405    # Method Not Allowed
-        || $_[0] == 410    # Gone
-        || $_[0] == 414    # Request-URI Too Large
-        || $_[0] == 451    # Unavailable For Legal Reasons
-        || $_[0] == 501    # Not Implemented
-      );
-}
+sub is_cacheable_by_default ($) { $_[0] &&
```
Comment (genio) [inline lib/HTTP/Status.pm]: Aesthetics here are debatable, but if you could, please revert this to the way that Perl::Tidy suggests. One of the goals is to eventually have everything tidied.  Thanks!

**Comment by vanHoesel** [comment]
i did a bit of tidying up ... no matter what, it will always be a piece of debate ... and yes, the only way to stop debates is calling the holy-grail perl-tidy ... and start debating on how to make the right settings to that

Also, added a few extra test, one of them making us alert that things were breaking

**Comment by genio** [comment]
We have a .perltidyrc in there to help.

```
$ perltidy lib/HTTP/Status.pm -o clean_status.pm
... copy in changes from clean_status.pm into lib/HTTP/Status.pm
$ rm clean_status.pm
```

Copy in the tidy changes for just the bit of code you're working on. Bits at a time. :)

**Comment by oalders** [comment]
Thanks everyone.  I'll merge and release now as that fixes the immediate issue.  If we need/want to re-introduce anything that was reverted, we can discuss that in a different issue.

**Pull request merged**